errands-routes.go

```diff
@@ -251,3 +252,50 @@ func (s *ErrandsServer) GetErrandsBy(fn func(*schemas.Errand) bool) []schemas.Er
 
 	return errands
 }
+
+func (s *ErrandsServer) clearErrands(c *gin.Context) {
+	var procErrand schemas.Errand
+
+	duration, err := time.ParseDuration(c.Param("duration"))
+	if err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{
+			"message": "Invalid Duration",
+			"error":   err.Error(),
+		})
+
+		return
+	}
+
+	threshold := time.Now().Add(0 - duration)
+	errands := make([]schemas.Errand, 0)
+
+	for _, itemObj := range s.Store.Items() {
+		item := itemObj.Object.(schemas.Errand)
+
+		var stoppedRunning int64
+
+		switch item.Status {
+		case "completed":
+			stoppedRunning = item.Completed
+		case "failed":
+			stoppedRunning = item.Failed
+		default:
+			continue
+		}
+
+		if stoppedRunning >= threshold.UnixNano() {
+			continue
+		}
+
+		errands = append(errands, item)
+	}
+
+	for _, errand := range errands {
+		s.deleteErrandByID(errand.ID)
+	}
+
+	c.JSON(http.StatusOK, gin.H{
+		"status":  "OK",
+		"results": procErrand,
```

Comment:
Doesn't look like `procErrand` is set anywhere, is this necessary?

Reply:
should be `errands` I think